src/sagas/SearchApiSagas.js

```diff
@@ -11,18 +11,66 @@ import { ERR_INVALID_ACTION, ERR_ACTION_VALUE_NOT_DEFINED } from '../utils/Error
 import { isValidAction } from '../utils/Utils';
 
 import {
+  EXECUTE_SEARCH,
   SEARCH_ENTITY_NEIGHBORS,
   SEARCH_ENTITY_NEIGHBORS_BULK,
   SEARCH_ENTITY_NEIGHBORS_FILTER,
   SEARCH_ENTITY_SET_DATA,
   SEARCH_ENTITY_SET_METADATA,
+  executeSearch,
   searchEntityNeighbors,
   searchEntityNeighborsBulk,
   searchEntityNeighborsWithFilter,
   searchEntitySetData,
   searchEntitySetMetaData,
 } from './SearchApiActions';
 
+/*
+ *
+ * SearchApi.executeSearch
+ * SearchApiActions.executeSearch
+ *
+ */
+
+function* executeSearchWorker(seqAction :SequenceAction) :Generator<*, *, *> {
+
+  if (!isValidAction(seqAction, EXECUTE_SEARCH)) {
+    return {
+      error: ERR_INVALID_ACTION
+    };
+  }
+
+  const { id, value } = seqAction;
+  if (value === null || value === undefined) {
+    return {
+      error: ERR_ACTION_VALUE_NOT_DEFINED
+    };
+  }
+
+  const response :Object = {};
+  const { searchConstraints } = value;
```

Comment:
Can you use `searchOptions` to match the other place in the file that already uses `searchOptions`?

Reply:
okay sure! the only reason i used `searchConstraints` was because that's what's used in `lattice-js` for this endpoint, but i guess it doesn't actually matter what the object being passed to the api is called.